### ISSUE TITLE
More efficient SaveResourceParameters

### ISSUE THREAD

**Issue by book987** [argoproj/argo-workflows#5658]
# Summary

Currently argo call `kubectl get` [for each output parameters](https://github.com/argoproj/argo-workflows/blob/v3.0.1/workflow/executor/resource.go#L340) in resource template. It can decrease to 1 time `kubectl get` call and `len(outputs.parameters)` times processing with same data, saving network traffic, and also preventing data inconsistency if something happen to the resource between the loop.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by terrytangyuan** [comment]
Good idea. It would be even better if we could switch to use k8s SDK for this as well instead of `kubectl`. Would you like to submit a PR for this?

**Comment by book987** [comment]
> Good idea. It would be even better if we could switch to use k8s SDK for this as well instead of `kubectl`. Would you like to submit a PR for this?

yeah, I'd like to.

**Comment by terrytangyuan** [comment]
`kubectl` removal: https://github.com/argoproj/argo-workflows/issues/7797

**Comment by Joibel** [comment]
`SaveResourceParameters` in `workflow/executor/resource.go` continues to issue one `kubectl get` per output parameter. The `jqFilter` path is the clearest case: it always requests `-o json` and evaluates the filter locally, so each JQ parameter re-fetches the identical document. A single fetch of the object as JSON, with JSONPath evaluated client-side (`k8s.io/client-go/util/jsonpath`, or `gjson` which this file already uses in `matchConditions`), would cover both the JSONPath and JQ branches and also removes the chance of different parameters observing different generations of the resource.

Note that #7797 removing the `kubectl` binary does not fix this — `runKubectl` now runs kubectl in-process, but each call is still a separate API-server read.
